src/iface/ethernet.rs

```diff
@@ -853,6 +875,44 @@ impl<'b, 'c> InterfaceInner<'b, 'c> {
         packet
     }
 
+    #[cfg(feature = "proto-ipv6")]
+    fn process_hopbyhop<'frame>(&mut self, sockets: &mut SocketSet, timestamp: Instant,
+                                ipv6_repr: Ipv6Repr, handled_by_raw_socket: bool,
+                                ip_payload: &'frame [u8]) -> Result<Packet<'frame>>
+    {
+        let hbh_pkt = Ipv6HopByHopHeader::new_checked(ip_payload)?;
+        let hbh_repr = Ipv6HopByHopRepr::parse(&hbh_pkt)?;
+        let buffer_len = hbh_repr.buffer_len();
+
+        let mut i = 2;
+        while i < buffer_len {
```

Comment:
I don't like this.

Reply:
How about making an `Ipv6OptionIterator`? Note that I'm not hung on this, and would be OK merging as is, but if you want, this would be how I improve it.

Reply:
:+1: yeah, I think that would be better. Should be pretty easy to implement.